### ISSUE TITLE
NaN exception when scaling pressable button parent to zero

### ISSUE THREAD

**Issue by julenka** [microsoft/MixedRealityToolkit-Unity#7874]
## Describe the bug
Filing on behalf of Matt Rubinstein whos will be making a github account soon.

We're using PressableButtonHololens2 and noticed that we get a NaN exception on transforms attached to MovingButtonVisuals/MovingButtonIconText/CompressableButtonVisuals when any parent up-chain of them has a 0 scale.
 

![image](https://user-images.githubusercontent.com/168492/81857721-810b2880-9517-11ea-9cbc-d061c84bce55.png)
 
We use a Hydration/Dehydration effect to scale up/down our prefabs on intro/outro which has them at 0 scale for a frame or two during the lifecycle. As a workaround, I modified our hydration effect to not go to exactly zero, but we still see this issue when values get too close to 0 scale.

## To reproduce

1. Set the scale of any parent of PressableButtonHoloLens 2 to zero
2. Observe NaN exceptions

## Release
MRTK 2.3

**Comment by julenka** [comment]
Assigned to @wiwei  for triage -- this falls under "fix things that are broken" and make things easy to configure from code" Probably not a regression in 2.4 and thus not release blocker.

**Comment by Zee2** [comment]
Hi!

This issue has been resolved in PR #7920. The explanation for this issue can be found there; a brief explanation is that zero-scaling the parent causes the `LocalToWorldScale` value to be undefined, which is what the button uses to adjust the `movingButtonVisuals`.

I was unable to actually trigger this error normally; when I set the parent scale to zero, I did not observe this exception until I quit the editor/exited play mode. (Upon exiting play mode, the above errors did occur). So, I'm not sure why those exceptions are thrown inconsistently! But regardless of when or how the error occurs, #7920 should definitely resolve it.

Cheers,

Finn
